mpas_analysis/shared/plot/plotting.py

```diff
@@ -27,6 +28,187 @@
+        axis labels
+
+    fileout : str
+        the file name to be written
+
+    linewidths : control line width
+
+    titleFontSize : int, optional
+        the size of the title font
+
+    figsize : tuple of float, optional
+        the size of the figure in inches
+
+    dpi : int, optional
+        the number of dots per inch of the figure
+
+    Author: Luke Van Roekel
+    Last Modified: 03/21/2017
+    """
+
+    plt.plot(f[2:-3], ninoSpectra[2:-3], 'k', linewidth=linewidths)
+    plt.plot(f[2:-3], redNoiseSpectra[2:-3], 'r', linewidth=linewidths)
+    plt.plot(f[2:-3], confidence95[2:-3], 'b', linewidth=linewidths)
+    plt.plot(f[2:-3], confidence99[2:-3], 'g', linewidth=linewidths)
+    plt.xlim(15, 1)
+
+    if titleFontSize is None:
+        titleFontSize = config.get('plot', 'titleFontSize')
+
+    axis_font = {'size': config.get('plot', 'axisFontSize')}
+    title_font = {'size': titleFontSize,
+                  'color': config.get('plot', 'titleFontColor'),
+                  'weight': config.get('plot', 'titleFontWeight')}
+    if title is not None:
+        plt.title(title, **title_font)
+    if xlabel is not None:
+        plt.xlabel(xlabel, **axis_font)
+    if ylabel is not None:
+        plt.ylabel(ylabel, **axis_font)
+    if fileout is not None:
+        plt.savefig(fileout, dpi=dpi, bbox_inches='tight', pad_inches=0.1)
+
+    if not config.getboolean('plot', 'displayToScreen'):
+        plt.close()
+
+
+def nino34_timeseries_plot(config, nino34Index, title, xlabel, ylabel,
+                           fileout, linewidths, calendar,
+                           titleFontSize=None, figsize=(15, 6), dpi=300,
+                           maxXTicks=20):
+    """
+    Plots the nino34 time series and power spectra in an image file
+     Parameters
+    ----------
+    config : instance of ConfigParser
+        the configuration, containing a [plot] section with options that
+        control plotting
+
+    nino34 : nino34 timeseries to plot
+
+    title : str
+        the title of the plot
+
+    xlabelTimeSeries, ylabelTimeSeries, xlabelSpectra, ylabelSpectra : str
+        axis labels
+
+    fileout : str
+        the file name to be written
+
+    lineStyles, lineWidths : list of str
+        control line style/width
+
+    titleFontSize : int, optional
+        the size of the title font
+
+    figsize : tuple of float, optional
+        the size of the figure in inches
+
+    dpi : int, optional
+        the number of dots per inch of the figure
+
+    maxXTicks : int, optional
+        the maximum number of tick marks that will be allowed along the x axis.
+        This may need to be adjusted depending on the figure size and aspect
+        ratio.
+
+    Author: Luke Van Roekel
+    Last Modified: 03/21/2017
+    """
+    plt.figure(figsize=figsize, dpi=dpi)
+
+    nino34Index = xr.DataArray.from_series(nino34Index)
+    y1 = nino34Index[2:-3]
```

Comment:
I believe if you switch(ed) the running mean to use xarray,  `nino34Index` is now an `xarray.DataSet` object.  If so, you should simply be able to slice it once: `nino34Index = nino34Index[2:3]` and get rid of all the other slicing below:
```
y1 = nino34Index.values
nt = len(y1)
time = nino34Index.Time.values
minDays = time[0]
maxDays = time[-1]
```